packages/node-opcua-data-model/source/diagnostic_info.ts

```diff
@@ -155,6 +168,36 @@ export interface DiagnosticInfoOptions {
     innerDiagnosticInfo?: DiagnosticInfo;
 }
 
+export enum DiagnosticInfo_ServiceLevelMask {
+    None = 0x00,
+    SymbolicId = 0x01,
+    LocalizedText = 0x02,
+    AdditionalInfo = 0x04,
+    InnerStatusCode = 0x08,
+    InnerDiagnostics = 0x10
+}
+
+export enum DiagnosticInfo_OperationLevelMask {
+    SymbolicId = 0x020,
+    LocalizedText = 0x040,
+    AdditionalInfo = 0x080,
+    InnerStatusCode = 0x0100,
+    InnerDiagnostics = 0x0200
+}
+
+type DiagnosticInfo_Mask = typeof DiagnosticInfo_ServiceLevelMask | typeof DiagnosticInfo_OperationLevelMask;
+
+export const RESPONSE_DIAGNOSTICS_MASK_ALL = 0x3FF;
+
+export function filterDiagnosticInfoLevel(returnDiagnostics: number, diagnostic: DiagnosticInfo | null, level: "service" | "operation"): DiagnosticInfo | null {
```

Comment:
I would replace:

```` 
level: "service" | "operation"
```` 
with
```` 
diagnosticInfoMask: DiagnosticInfo_Mask
```` 

so that
```` 
 const enumeration = level === "service" ? DiagnosticInfo_ServiceLevelMask : DiagnosticInfo_OperationLevelMask;
    return DiagnosticInfo.filterForResponse(diagnostic, returnDiagnostics, enumeration);
```` 
becomes
````
    return DiagnosticInfo.filterForResponse(diagnostic, returnDiagnostics, diagnosticInfoMask);
````